
From: Ryder Lee <ryder.lee@mediatek.com>
Date: Sat, 18 Feb 2023 01:50:05 +0800
Subject: [PATCH] wifi: mac80211: introduce
 ieee80211_refresh_tx_agg_session_timer()

This allows low level drivers to refresh the tx agg session timer, based on
querying stats from the firmware usually. Especially for some mt76 devices
support .net_fill_forward_path would bypass mac80211, which leads to tx BA
session timeout for certain clients.

Signed-off-by: Ryder Lee <ryder.lee@mediatek.com>
---

--- a/drivers/net/wireless/mediatek/mt76/mt7915/eeprom.c
+++ b/drivers/net/wireless/mediatek/mt76/mt7915/eeprom.c
@@ -206,6 +206,9 @@ void mt7915_eeprom_parse_hw_cap(struct mt7915_dev *dev,
 
                if (!is_mt7986(&dev->mt76))
                        nss_max = 2;
+
+               if (!nss)
+                       nss = nss_max;
        }
 
        nss = min_t(u8, min_t(u8, nss_max, nss), path);
